### PR TITLE
Skip tests that require root when not root

### PR DESCRIPTION
**Pull request by schmichael**
Also skip Chown on allocdir migration on Windows and when non-root.
Windows doesn't support it, and it will always fail as a non-root user.

### CODE REVIEW

File: client/alloc_watcher.go [hunk 1]
```diff
@@ -517,9 +518,14 @@ func (p *remotePrevAlloc) streamAllocDir(ctx context.Context, resp io.ReadCloser
 				f.Close()
 				return fmt.Errorf("error chmoding file %v", err)
 			}
-			if err := f.Chown(hdr.Uid, hdr.Gid); err != nil {
-				f.Close()
-				return fmt.Errorf("error chowning file %v", err)
+
+			// Can't change owner if not root. Returns false on
```
Comment (preetapan) [inline client/alloc_watcher.go]: Why is there an increased scope of this commit beyond only skipping tests? IIUC This changes the behavior of alloc runner in a subtle way on windows, from returning an error to skipping the chown step. If the chown step is not optional, won't it error further downstream?

Comment (schmichael) [reply client/alloc_watcher.go]: In this case the test didn't need fixing; the code shouldn't require root. The code was requiring Nomad to run as root on Linux -- which is our recommendation but not a requirement. The code also totally prevented sticky migrations from working on Windows where [`os.Chown` always returns an error.](https://golang.org/pkg/os/#Chown)

The only negative side effect I can think of is if an alloc was migrated from a node running as root and relied on multiple file ownerships, and was migrated to a node not running as root where those ownerships would suddenly be lost.

I can switch to only skipping chown on Windows if we'd rather require root for safety.

File: client/alloc_watcher.go [hunk 2]
```diff
@@ -517,9 +518,14 @@ func (p *remotePrevAlloc) streamAllocDir(ctx context.Context, resp io.ReadCloser
...
+			// Windows as Chown always errors there.
+			if syscall.Geteuid() == 0 {
```
Comment (dadgar) [inline client/alloc_watcher.go]: You should cache this as you will potentially be making a lot of syscalls!

File: client/alloc_watcher.go (hunk 1 above)
Comment (dadgar) [reply client/alloc_watcher.go]: Hmm this should probably make it into 0.7.1 final if that is the case

Comment (schmichael) [reply client/alloc_watcher.go]: I should clarify one point:

You can chown as a non-root user only if it's a no-op (you're chown'ing a file you own to be owned by you). So as long as Nomad is running as the same non-root user on every node migrations will work without Nomad being run as root.

So this mainly affects Windows as I would assume most people running Nomad as a non-root user create a `nomad` user.

File: client/alloc_watcher.go (hunk 2 above)
Comment (schmichael) [reply client/alloc_watcher.go]: Good call. Fixed!

File: client/alloc_watcher.go (hunk 1 above)
Comment (preetapan) [reply client/alloc_watcher.go]: I would vote for making the Windows condition very explicit then. This avoids us from forgetting the subtlety mentioned above.

**Comment by schmichael** [comment]
Test failure is unrelated.

**Pull request merged**